### ISSUE TITLE
with respect to the file format in example in multi_relation_example.sh

### ISSUE THREAD

**Issue by bastia1989** [facebookresearch/StarSpace#234]
The train file format is:
`entity\t relation\t__label__ entity`, why it has the prefix `__label__` before the tail entity rather than the relation?

**Comment by ledw** [comment]
@bastia1989 hi, for the link prediction task, the model predicts entity instead of relation, thus the entities are the labels in this task. For the case the head/tail entity has multiple words, do you want to featurize the entities by its words (so you learned word embeddings at the end), or do you want to learn an embedding for each entity? For the former case, please use `-fileFormat labelDoc`. For the later, you can use the default setting of fileFormat.

**Comment by bastia1989** [comment]
> @bastia1989 hi, for the link prediction task, the model predicts entity instead of relation, thus the entities are the labels in this task. For the case the head/tail entity has multiple words, do you want to featurize the entities by its words (so you learned word embeddings at the end), or do you want to learn an embedding for each entity? For the former case, please use `-fileFormat labelDoc`. For the later, you can use the default setting of fileFormat.

Thanks for the reply. In my case, I want to learn the embedding of each word of the entity, and in the prediction I want to predict the **RELATION** given head/tail entities. So for my case, I should add `__label__` before the relation, right? Since here in my case, I don't have multiple but sole relationship, so I think I don't need `-fileFormat labelDoc`. Meanwhile, I don't know that why ` '' ` or `null` is also embedded in the result tsv file (the first column of the tsv file contains a NULL element).

**Comment by bastia1989** [comment]
> @bastia1989 hi, for the link prediction task, the model predicts entity instead of relation, thus the entities are the labels in this task. For the case the head/tail entity has multiple words, do you want to featurize the entities by its words (so you learned word embeddings at the end), or do you want to learn an embedding for each entity? For the former case, please use `-fileFormat labelDoc`. For the later, you can use the default setting of fileFormat.

By the way, in my case, there is one relation called 'mad of', so I tagged it as `__label__made of`, I am not sure if this is the correct way. It seems that even I don't set `-fileFormat labelDoc`, in the final result I have `__label__made` and `of` both exists, which looks like they are treated as two words rather than an entity.

Meanwhile, I believe we may need the vectors of head/tail entities and relationships of different size (say, 128 for head/tail, and 64 for relationship). Is that workable for starspace?

Thanks.

**Comment by bastia1989** [comment]
anyupdates?

**Comment by ledw** [comment]
@bastia1989 Hi, sorry for the delay in replying. Yes, you should add __label__ before relation in your case, and no need to use the labelDoc fileFormat. However, you need to do some preprocessing in your input, as StarSpace splits by space. One thing you can do is to change 'made of' to be something like 'made-of', i.e. add a special separate character between 'made' and 'of' so that it will be treated as a phrase in StarSpace.

**Comment by bastia1989** [comment]
> @bastia1989 Hi, sorry for the delay in replying. Yes, you should add **label** before relation in your case, and no need to use the labelDoc fileFormat. However, you need to do some preprocessing in your input, as StarSpace splits by space. One thing you can do is to change 'made of' to be something like 'made-of', i.e. add a special separate character between 'made' and 'of' so that it will be treated as a phrase in StarSpace.

Thanks for the answer! Then can we control the dimension of the vectors of entity and relation differently?

**Comment by ledw** [comment]
@bastia1989 no, you need to use the same dimension for vectors of entity and relations.
